### DISCUSSION TITLE
Installation not working with environment database connection

### DISCUSSION THREAD

**Discussion by Broatcast** [codeigniter4/shield#937] [Ideas]
When the database connection is setup in environment via the webserver. Spark is not able to get this information and is failing the installation/dbmigration. 

Helpfull could be detecting the empty/default database configuration and ask the user for the credentials in this process.

**Comment by sammyskills** [comment]
Quick question:

_Have you renamed or created a `.env` file and NOT just `env` ?_

**Reply by Broatcast** [reply]
Environment configuration via webserver... means it gets its information from the http/php instance. For example nginx:

```
location ~ "^(.+\.php)($|/)" {
    fastcgi_split_path_info ^(.+\.php)(.*)$;

    fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
    fastcgi_param SCRIPT_NAME $fastcgi_script_name;
    fastcgi_param PATH_INFO $fastcgi_path_info;
    fastcgi_param SERVER_NAME $host;

    # CI ENVIRONMENTS
    fastcgi_param CI_ENVIRONMENT "development";
    fastcgi_param database.default.database "dbname";
    fastcgi_param database.default.password "dbpass";
    fastcgi_param database.default.username "dbuser";
    fastcgi_param database.default.DBDriver "MySQLi";
    fastcgi_param database.default.DBPrefix "";
    fastcgi_param app.baseURL "http://dev.local/";

    if ($uri !~ "^/uploads/") {
      fastcgi_pass unix:/run/php/php8.2-fpm.sock;
    }
    include fastcgi_params;
  }
```

In such cases there are no informations stored in the .env files, which is the whole point for this kind of feature.

**Reply by sammyskills** [reply]
Does CI detect the environment configurations set at the webserver level?

**Reply by kenjis** [reply]
`php` command is not nginx. 
So the environment variables for nginx cannot be read by php (spark).

**Reply by Broatcast** [reply]
I know, thats why this isnt detected and i got the idea for a check while installing and questioning the operator to to enter them manual ... like questioning for the mail addresses for example.

**Reply by kenjis** [reply]
Okay, I got why you did post to "Ideas".

But not only in installation, but also all spark commands that depend on the variables do not work in your configuration.

> Requirements
> A created database that you can access via the Spark CLI script
https://codeigniter4.github.io/shield/getting_started/install/#requirements

**Reply by Broatcast** [reply]
I know that its mentioned, i just have the idea its maybe helpfull for deployment to catch this case of configurations, instead of forcing the user to manualy edit the database configuration for the installation process and redo it afterwards.

More nicely would be to detect if the database config is default/empty and ask to provide for credentials in the setup process.

**Reply by kenjis** [reply]
Why don't you configure the environment variables for shell?

```console
$ php spark db:table

CodeIgniter v4.4.3 Command Line Tool - Server Time: 2023-10-31 22:15:28 UTC+00:00

+-----------+----------+----------+----------+----------+------+
| hostname  | database | username | DBDriver | DBPrefix | port |
+-----------+----------+----------+----------+----------+------+
| localhost |          |          | MySQLi   |          | 3306 |
+-----------+----------+----------+----------+----------+------+


[mysqli_sql_exception]

You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1

at SYSTEMPATH/Database/MySQLi/Connection.php:306
```

```console
$ cat test.env 
export database_default_database=ci4
export database_default_username=root
export database_default_password=root
```
```console
$ source test.env
$ php spark db:table

CodeIgniter v4.4.3 Command Line Tool - Server Time: 2023-10-31 22:16:18 UTC+00:00

+-----------+----------+----------+----------+----------+------+
| hostname  | database | username | DBDriver | DBPrefix | port |
+-----------+----------+----------+----------+----------+------+
| localhost | ci4      | root     | MySQLi   |          | 3306 |
+-----------+----------+----------+----------+----------+------+

Database has no tables!
```
